test/e2e/apps/cronjob.go

```diff
@@ -280,6 +286,140 @@ var _ = SIGDescribe("CronJob", func() {
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(len(clusterCJs.Items), 1, "filtered list should have 1 items")
+
+		ginkgo.By("cluster-wide watching")
+		framework.Logf("starting watch")
+		_, err = clusterCJClient.Watch(context.TODO(), metav1.ListOptions{ResourceVersion: cjs.ResourceVersion, LabelSelector: "special-label=" + f.UniqueName})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("patching")
+		patchedCronJob, err := cjClient.Patch(context.TODO(), createdCronJob.Name, types.MergePatchType,
+			[]byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(patchedCronJob.Annotations["patched"], "true", "patched object should have the applied annotation")
+
+		ginkgo.By("updating")
+		var cjToUpdate, updatedCronJob *batchv1beta1.CronJob
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			cjToUpdate, err = cjClient.Get(context.TODO(), createdCronJob.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			cjToUpdate.Annotations["updated"] = "true"
+			updatedCronJob, err = cjClient.Update(context.TODO(), cjToUpdate, metav1.UpdateOptions{})
+			return err
+		})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(updatedCronJob.Annotations["updated"], "true", "updated object should have the applied annotation")
+
+		framework.Logf("waiting for watch events with expected annotations")
+		for sawAnnotations := false; !sawAnnotations; {
+			select {
+			case evt, ok := <-cjWatch.ResultChan():
+				framework.ExpectEqual(ok, true, "watch channel should not close")
+				framework.ExpectEqual(evt.Type, watch.Modified)
+				watchedCronJob, isCronJob := evt.Object.(*batchv1beta1.CronJob)
+				framework.ExpectEqual(isCronJob, true, fmt.Sprintf("expected CronJob, got %T", evt.Object))
+				if watchedCronJob.Annotations["patched"] == "true" {
+					framework.Logf("saw patched and updated annotations")
+					sawAnnotations = true
+					cjWatch.Stop()
+				} else {
+					framework.Logf("missing expected annotations, waiting: %#v", watchedCronJob.Annotations)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+
+		// /status subresource operations
+		ginkgo.By("patching /status")
+		// we need to use RFC3339 version since conversion over the wire cuts nanoseconds
+		now1 := metav1.Now().Rfc3339Copy()
+		cjStatus := batchv1beta1.CronJobStatus{
+			LastScheduleTime: &now1,
+		}
+		cjStatusJSON, err := json.Marshal(cjStatus)
+		framework.ExpectNoError(err)
+		patchedStatus, err := cjClient.Patch(context.TODO(), createdCronJob.Name, types.MergePatchType,
+			[]byte(`{"metadata":{"annotations":{"patchedstatus":"true"}},"status":`+string(cjStatusJSON)+`}`),
+			metav1.PatchOptions{}, "status")
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(patchedStatus.Status.LastScheduleTime, &now1, "patched object should have the applied lastScheduleTime status")
```

Comment:
This fails CI
```
test/e2e/apps/cronjob.go:289
Feb 22 18:38:32.590: patched object should have the applied lastScheduleTime status
Expected
    <*v1.Time | 0xc0026467e0>: {
        Time: 2021-02-22T18:38:32Z,
    }
to equal
    <*v1.Time | 0xc0003ce640>: {
        Time: 2021-02-22T18:38:32Z,
    }
test/e2e/apps/cronjob.go:387
```
looks like the object reference is throwing it off here? Maybe looking for a semantic equal

Reply:
Switched to `metav1.Time.Equal` which should be better.